src/backend/distributed/citus--7.1-3--7.1-4.sql

```diff
@@ -0,0 +1,54 @@
+/* citus--7.1-3--7.1-4 */
+
+CREATE TYPE citus.shard_transfer_mode AS ENUM (
+   'require_replica_identity',
+   'force_logical',
+   'block_writes'
+);
+
+SET search_path = 'pg_catalog';
+
+DROP FUNCTION master_move_shard_placement(bigint, text, integer, text, integer);
+DROP FUNCTION master_copy_shard_placement(bigint, text, integer, text, integer, bool);
+
+CREATE OR REPLACE FUNCTION master_move_shard_placement(
+	shard_id bigint,
+	source_node_name text,
+	source_node_port integer,
+	target_node_name text,
+	target_node_port integer,
+	shard_transfer_mode citus.shard_transfer_mode default 'require_replica_identity')
+RETURNS void LANGUAGE C STRICT
+AS 'MODULE_PATHNAME', $$master_move_shard_placement$$;
+
+COMMENT ON FUNCTION master_move_shard_placement(
+	shard_id bigint,
+	source_node_name text,
+	source_node_port integer,
+	target_node_name text,
+	target_node_port integer,
+	shard_transfer_mode citus.shard_transfer_mode)
+IS 'move a shard from a the source node to the destination node';
+
+CREATE FUNCTION master_copy_shard_placement(
+	shard_id bigint,
+	source_node_name text,
+	source_node_port integer,
+	target_node_name text,
+	target_node_port integer,
+	do_repair bool DEFAULT true,
+	transfer_mode citus.shard_transfer_mode default 'require_replica_identity')
+RETURNS void
+LANGUAGE C STRICT
+AS 'citus', $$master_copy_shard_placement$$;
+
+COMMENT ON FUNCTION master_copy_shard_placement(shard_id bigint,
```

Comment:
I think my only comment would be that we're not supporting logical replication for shard repair functionality and actually ignoring the  `shard_transfer_mode ` flag. 

We could potentially error out on `master_copy_shard_placement()` when `doRepair = true` and `shard_transfer_mode != block_writes`. But, that'd break quite a few regression tests and potentially breaks the user experience. Still, it'd be useful to at least have some `LOG` messages for that.

What is your thought on that?

Reply:
I've added an error when using `force_logical`. Maybe `require_replica_identity` is misnamed and should be something more generic, e.g. `auto`?

Reply:
Actually, that makes sense to me. `auto` in the sense that it tries to pick the best alternative and errors under certain conditions where it cannot pick the best. In fact, `require_replica_identity` is hard to understand for users.

Reply:
Alright, changed to `auto`